src/main/java/org/bananarama/BananaRama.java

```diff
@@ -41,6 +42,11 @@
     
     private final StripedLock slock;
     private final Map<Class<?>, Adapter<?>> adapters;
+    private static final Map<Class<?>, Class<? extends Adapter>> customAdapters = new ConcurrentHashMap<>();
+    
+    public static void registerAdapter(Class<? extends Adapter> adapter, Class forEntiy) {
```

Comment:
BAD IDEA. This can easily generate unpredictable behavior at runtime as it breaks correlation between how class are annotated and how they are loaded.